Automattic/wp-calypso
# DocumentHead: Set Meta and Link (instead of appending)

Replace `addDocumentHeadMeta` and `addDocumentHeadLink` by `setDocumentHeadMeta` and `setDocumentHeadLink`, respectively.

Just adding to them (without providing a clear/reset action) is error-prone as it can lead to duplication of `meta` and `link` tags -- right now, I'm seeing this in the Page Source for wordpress.com/themes:

![image](https://cloud.githubusercontent.com/assets/96308/25802052/f1e176bc-33f0-11e7-8268-956bd5a1a0d1.png)

This PR purports to fix that bug.

To test:
* Logged-out, check the Page Source for `http://calypso.localhost:3000/themes`. Compare to production, and make sure that no `meta` or `link` tags were lost (try Cmd+F `og:`)
* Same for a theme sheet, e.g. `http://calypso.localhost:3000/theme/mood`
* Unfortunately, I wasn't able to repro the original bug locally (I tried reloading `/themes` both logged-in and out), so I can't demonstrate that this PR fixes it. You'll have to trust the diff (but you can at least verify that it doesn't introduce any regressions.)
